tests/Feature/ImportReportTest.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Tests\Feature;
+
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Foundation\Testing\WithFaker;
+use Tests\TestCase;
+use App\Services\ImportReport;
+use Illuminate\Support\Facades\Storage;
+use App\Models\User;
+use App\Models\ImportMeta;
+use App\Models\Mismatch;
+
+class ImportReportTest extends TestCase
+{
+    use RefreshDatabase, WithFaker;
+    /**
+     * @return void
+     */
+    public function test_generate_csv_writes_to_path(): void
+    {
+        $user = User::factory()->uploader()->create();
+        $import = ImportMeta::factory()->for($user)->create(
+            ['status' => 'completed']
+        );
+        Mismatch::factory()->for($import)->create();
```

Comment:
I think at this point I would have added at least two mismatches, one pending and one reviewed, if only to show that the "% completed" calculation is done correctly.

Reply:
done, I even added three so that it would b 33.33% vs 66.67% :D